src/XRP/issued-currency-client.ts

```diff
@@ -126,6 +132,58 @@ export default class IssuedCurrencyClient {
     return trustLines
   }
 
+  /**
+   * Returns information about the total balances issued by a given account,
+   * optionally excluding amounts held by operational addresses.
+   * @see https://xrpl.org/issuing-and-operational-addresses.html
+   *
+   * @param account The account for which to retrieve balance information, encoded as an X-Address.
+   * @param accountsToExclude (Optional) An array of operational addresses to exclude from the balances issued, encoded as X-Addresses.
+   * @see https://xrpaddress.info/
+   * @returns A GatewayBalances object containing information about an account's balances.
+   */
+  public async getGatewayBalances(
+    account: string,
+    accountsToExclude: Array<string> = [],
+  ): Promise<GatewayBalances> {
+    // check issuing account for X-Address format
+    const classicAddress = XrpUtils.decodeXAddress(account)
+    if (!classicAddress) {
+      throw XrpError.xAddressRequired
+    }
+
+    // check excludable addresses for X-Address format
+    const classicAddressesToExclude: Array<string> = []
+    accountsToExclude.map((xAddress) => {
+      const excludeClassicAddress = XrpUtils.decodeXAddress(xAddress)
+      if (!excludeClassicAddress) {
+        throw XrpError.xAddressRequired
+      }
+      classicAddressesToExclude.push(excludeClassicAddress.address)
+    })
```

Comment:
you can just do

```
const classicAddressesToExclude = accountsToExclude.map(() => {
  // code code code
})
```

`map` returns an `Array` already

Reply:
thanks for teaching me javascript ted